core/src/main/java/org/apache/druid/java/util/emitter/core/EventMap.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.java.util.emitter.core;
+
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * EventMap is a hash map implementation. It can be safely serialzed to JSON using Jackson serializer as it
+ * respects the polymorphic annotations on entires (unlike standard Map). The example of polymorphic class is a query
+ * interface, where different native query types are resolved by additional field called "queryType".
+ * This implementation ensures that the annotation on the values are respected during serialization.
+ */
+@JsonSerialize(using = EventMapSerializer.class)
+public class EventMap extends HashMap<String, Object>
```

Comment:
@BartMiki @FrankChen021 : I just noticed that we're extending `HashMap` instead of a `Map`. This would mean that any of the consumers extending `PublicApi` like `AlertEvent`, `RequestLogEvent` or `ServiceMetricEvent` can't use any other forms of `Map` in their implementations other than `HashMap`.
Further, even if someone is using `HashMap<String, Object>` already, they would need to cast the map to an `EventMap` which I don't think is correct.
So, the correct solution for the developers would be to change their implementation to use `EventMap` itself which again break compatibility. Are we ok with doing this change?

Reply:
I followed the GH issue thread and there is seems that this was discussed and we're ok with this change (https://github.com/apache/druid/issues/12912#issuecomment-1221918198). will wait for confirmatiom from Frank

Reply:
@rohangarg Since we have changed the signature of `toMap()` in the `Event` interface and all its implementations in this PR, I think there won't be compatibility problems.

Reply:
Thanks for the explanations Bart and Frank! I agree that it won't break compatibility since the callers of the `Event` impls. can still cast the `EventMap` returned by `toMap()` to a `Map<String, Object>` and that's the only contract for `PublicApi` annotated interfaces.